### DISCUSSION TITLE
Using Angular CDK

### DISCUSSION THREAD

**Discussion by elenagarrone** [Legal-and-General/canopy#392] [Q&A] [answered]
Hi everyone,
I was wondering what's everyone's opinion of using some of the Angular Material CDK features in Canopy? For example I'm currently working on a modal component where I would need to use the focus trap. Are we happy to import it from the CDK or we prefer to have those things built in our own codebase?

I don't see any down side in using the cdk. Does anyone disagree?

**Comment by andy-polhill** [comment, marked as answer]
I always imagined we would end up needing the CDK for certain things like floating elements, if it wasn't the CDK it would likely be another 3rd party library. I'm happy for us to pull it in and take a look.

**Comment by paul-request** [comment]
Yeah makes sense to bring it in now if there is a requirement for that sort of thing

**Comment by owensgit** [comment]
I agree with @paul-request and @andy-polhill, let's give it a go 👍
